nmstatectl/nmstatectl.py

```diff
@@ -57,9 +72,29 @@ def main():
         parser.print_usage()
         return errno.EINVAL
     args = parser.parse_args()
+
+    try:
+        _setup_logging(args.log_level)
+    except ValueError:
+        return os.EX_USAGE
+
     return args.func(args)
 
 
+def _setup_argument_log_level(parser):
+    """
+    Setup the log level argument.
+    Default to the value passed through the environment variable.
+    """
+    log_level = os.getenv(NmstateEnvVar.LOG_LEVEL, LOG_LEVEL_DEFAULT)
+    parser.add_argument(
+        '--log-level',
```

Comment:
Why do you prefer setting the log-level instead of using a reasonable log level by default and allowing to increase/decrease it with -q/-v?

Reply:
Per what I know, `-v` or `-q` are usually controlling stdout/sdterr and not the loggers. We can log by default to a file and use `-q/v` to control if we want it on the console as well or not.

Reply:
Currently, even though the argument is called `log-level`, it actually configures how much information is outputted at the console because nmstatectl is just using the logging module to output to the console. So you could also add an extra feature to log to a logfile (which I think is not useful for nmstatectl) that could be controlled as well.

Reply:
You are pointing our a consequence of the change, not the change itself. The commit message and the change implementation is controlling the *log-level*, it does not mention anything about changing what is shown on the console.

Attempting to couple `-v/q` to the log-level is IMO improper, imposing limitations which are not useful.
`-q` is usually used to redirect stdout/stderr to `/dev/null`.
`-v` is usually used to choose what to display on the console, including the logging but not limited to it. For example, should the state *diff* be presented, at what depth, should the final state be presented at the end, etc...

Reply:
I don't think the user of `nmstatectl` care about the difference between `controling stdout/stderr` vs `logging`.

For me, I use `-vvv` or `-d9` to get all debug output from a CLI and I don't want to type things like `--log-level NOTSET` where I need to know which log level can get all the logs.

Reply:
You are using a reasoning of "not needed", where the original reasoning is both common usage (see other projects) and flexibility to separate between verbose or quite output and the logging mechanism.

You both are suggesting to use a non standard translation between `-vvvvv` to the log level and mix the ability to output diffs or other output which has nothing to do with the logging in one argument/place.
I do not understand how this is making any sense.

Reply:
My argument is user of nmstatectl does not care about log level and `-q/-v` is enough for current design.

We can add this complex-long-typing argument `--log-level` when a actual user asked about it.

Reply:
Sounds like a deadlock to me.
I'm ok with `-q` / `-v` controlling the stdout/stderr, not with it controlling the logging level.
So if you want a solution that does not print anything to the console, please propose a different change, which redirects logging to `/dev/null` or to the logging service.